src/algebra/add_submonoid_closure.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import ring_theory.subsemiring
+
+/-!
+#  Equality between `add_submonoid.closure S` and `subsemiring.closure S`
+
+Let `R` be a semiring and let `S` be a multiplicative submonoid of `R`.
+The additive closure of `S` is a subsemiring of `R`.  If `R` is commutative, then so
+is the additive closure of `S`.
+-/
+variables {R : Type*} {a b : R}
+
+section semiring
+
+variables [semiring R] {S : submonoid R}
+
+/-- The product of an element of the additive closure of a multiplicative submonoid `S`
+and an element of `S` is contained in the additive closure of `S`. -/
+lemma add_submonoid.closure_mem_mul (ha : a ∈ add_submonoid.closure (S : set R)) (hb : b ∈ S) :
```

Comment:
I think `submonoid.mul_mem_add_closure` _might_ be a better name for this lemma, along with making `S` implicit to allow dot notation. What do you think?

Reply:
I like your suggestion, although I also think that this lemma is just a step in the proof of the fact that the product of two elements of the `add_closure` is in the add closure and may not be so useful.  If, on the other hand, you think that it is actually useful, besides the suggested name change and implicitization, I can also add the lemma with the product on the other side.

Reply:
Just to make sure that I understand: you were thinking of making `S`*ex*plicit, to allow dot-notation, right?

Reply:
That's a good counter-argument; indeed if this is just some auxiliary work, then maybe it should be `private`

Reply:
Well, I still think that you may have an term of the (multiplicative) submonoid and a term of of its additive closure around and may be wanting to multiply them together.  Besides, I am not so sure that I like the `private` tag: a lemma is a lemma after all!